polaris-core/src/main/java/org/apache/polaris/core/persistence/BasePersistence.java

```diff
@@ -37,6 +37,11 @@
  * defining the internal data model for Polaris, and which defines the basis for the RBAC model
  * provided by Polaris.
  *
+ * <p>Each method in this interface must be atomic, meaning that write operations must either fully
+ * succeed with all changes applied, or fail entirely without any side effects or partial updates.
+ * Read operations must return a consistent view of the data as it existed at a specific point in
```

Comment:
What is this "specific point in time"?

Do all reads have to be based on the same point in time?

Reply:
+1, if we have such a strong requirement then I think we should be clear about what that point in time is.

Also, what is a "read operation"? How does that interact with pagination?

Reply:
Added more accurate description that a snapshot isolation for read in the new commit.